### PR TITLE
feat(testing-auth): Add tests for auth

### PR DESCRIPTION
**Pull request by nicugnm**
Features:
- Add unit tests for every auth class.

### CODE REVIEW

File: src/test/java/ro/unibuc/triplea/infrastructure/auth/fixtures/TokenFixtures.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import ro.unibuc.triplea.domain.auth.model.enums.TokenType;
+
+public class TokenFixtures {
+
+    public static Token revokedNotExpiredToken(String token) {
+        return new Token(1, token, TokenType.BEARER, true, false, UserFixtures.user("username"));
+    }
+
+    public static Token validToken(String token) {
+        return new Token(1, token, TokenType.BEARER, false, false, UserFixtures.user("username"));
```
Comment (RaduADumitru) [inline src/test/java/ro/unibuc/triplea/infrastructure/auth/fixtures/TokenFixtures.java:13]: Good approach of using text fixtures, I learned something new today

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -16,7 +16,7 @@
 
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.6-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.5-bin.zip
 networkTimeout=10000
```
Comment (RaduADumitru) [inline gradle/wrapper/gradle-wrapper.properties:20]: Could conflict with config changes on main

File: build.gradle [hunk 3]
```diff
@@ -118,4 +119,8 @@ tasks.register('testE2E') {
 tasks.named('asciidoctor') {
 	inputs.dir snippetsDir
 	dependsOn test
+}
+
+test {
+	useJUnitPlatform()
```
Comment (RaduADumitru) [inline build.gradle:125]: Could conflict with config changes on main; junit was also added there

File: build.gradle [hunk 4]
```diff
@@ -46,6 +46,7 @@ dependencies {
 	compileOnly 'org.projectlombok:lombok'
 	runtimeOnly 'org.postgresql:postgresql'
 	annotationProcessor 'org.projectlombok:lombok'
+	testImplementation 'org.projectlombok:lombok:1.18.28'
```
Comment (RaduADumitru) [inline build.gradle:49]: Could conflict with config changes on main

**Review by RaduADumitru — CHANGES_REQUESTED** [review]
Nice approach for the tests themselves, but please do a merge from main before merging the PR, and ensure there are no conflicts in configuration and everything works fine after - some aspects would be duplicated, like adding Junit which we've done in the meantime

**Comment by nicugnm** [comment]
Merge with main is already done, check commits

File: gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (nicugnm) [reply gradle/wrapper/gradle-wrapper.properties]: rollbacked

**Review by RaduADumitru — APPROVED** [review]
LGTM

**Pull request merged**